hazelcast/src/main/java/com/hazelcast/version/Version.java

```diff
@@ -0,0 +1,162 @@
+    }
+
+    public Version(int major, int minor, int patch) {
+        this.major = (byte) major;
+        this.minor = (byte) minor;
+        this.patch = (byte) patch;
+    }
+
+    public Version(String version) {
+        String[] tokens = version.split("\\.");
+        this.major = Byte.valueOf(tokens[0]);
+        this.minor = Byte.valueOf(stripPostfix(tokens[1]));
+        if (tokens.length > 2) {
+            this.patch = Byte.valueOf(stripPostfix(tokens[2]));
+        }
+    }
+
+    private static String stripPostfix(String token) {
+        int hyphenIndex = token.indexOf("-");
+        return hyphenIndex >= 0 ? token.substring(0, hyphenIndex) : token;
+    }
+
+    public byte getMajor() {
+        return major;
+    }
+
+    public byte getMinor() {
+        return minor;
+    }
+
+    public byte getPatch() {
+        return patch;
+    }
+
+    public boolean isUnknown() {
+        return this.equals(UNKNOWN);
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+        Version that = (Version) o;
+        if (major != that.major) {
+            return false;
+        }
+        if (minor != that.minor) {
+            return false;
+        }
+        return patch == that.patch;
+    }
+
+    @Override
+    public int hashCode() {
+        int result = (int) major;
+        result = 31 * result + (int) minor;
+        result = 31 * result + (int) patch;
+        return result;
+    }
+
+    @Override
+    public String toString() {
+        return major + "." + minor + "." + patch;
+    }
+
+    @Override
+    public void writeData(ObjectDataOutput out) throws IOException {
+        out.writeByte(major);
+        out.writeByte(minor);
+        out.writeByte(patch);
+    }
+
+    @Override
+    public void readData(ObjectDataInput in) throws IOException {
+        this.major = in.readByte();
+        this.minor = in.readByte();
+        this.patch = in.readByte();
+    }
+
+    public static Version of(int major, int minor, int patch) {
+        return new Version(major, minor, patch);
+    }
+
+    public static Version of(String version) {
+        return new Version(version);
+    }
+
+    @Override
+    public int getFactoryId() {
+        return ClusterDataSerializerHook.F_ID;
+    }
+
+    @Override
+    public int getId() {
+        return ClusterDataSerializerHook.VERSION;
+    }
+
+    @Override
+    @SuppressWarnings("checkstyle:booleanexpressioncomplexity")
+    public int compareTo(Version otherVersion) {
+        int thisVersion = (major << 24 & 0xff0000) | ((minor << 16) & 0xff00) | (patch & 0xff);
+        int thatVersion = (otherVersion.major << 24 & 0xff0000) | (otherVersion.minor << 16 & 0xff00)
+                | (otherVersion.patch & 0xff);
+        return (thisVersion > thatVersion) ? 1 : (thisVersion == thatVersion ? 0 : -1);
```

Comment:
Nested ternary operations are a pain to read. Can you convert the outer one to an if else?